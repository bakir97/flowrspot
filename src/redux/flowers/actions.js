import axios from "axios";
import { devKeys } from "../../globals/keys";
import { LOADING_FLOWERS, SUCCESS_FLOWERS } from "../constants";
export const getFlowers = () => async dispatch => {
  try {
    dispatch(loadingFlowers(true));
    const flowers = await axios.get(`${devKeys.api}/flowers`, {
      headers: { Authorization: devKeys.getFlowersAuth }
    });
    dispatch(successFlowers(flowers.data.flowers));
    dispatch(loadingFlowers(false));
  } catch (error) {
    dispatch(loadingFlowers(false));
  }
};
export const searchFlowers = search => async dispatch => {
  try {
    dispatch(loadingFlowers(true));

    const flowers = await axios.get(
      `${devKeys.api}/flowers/search?query=${search}`,
      {
        headers: { Authorization: devKeys.flowersSearchAuth }
      }
    );
    dispatch(successFlowers(flowers.data.flowers));
    dispatch(loadingFlowers(false));
  } catch (error) {
    console.log(error);

    dispatch(loadingFlowers(false));
  }
};
export const loadingFlowers = payload => ({
  type: LOADING_FLOWERS,
  payload
});
export const successFlowers = payload => ({
  type: SUCCESS_FLOWERS,
  payload
});
