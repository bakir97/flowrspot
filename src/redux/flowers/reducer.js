import { LOADING_FLOWERS, SUCCESS_FLOWERS, ADD_FAV } from "../constants";
const initialState = {
  flowers: [],
  loading: false
};

export default (state = initialState, { type, payload }) => {
  switch (type) {
    case LOADING_FLOWERS:
      return { ...state, loading: payload };
    case SUCCESS_FLOWERS:
      return { ...state, flowers: payload };
    case ADD_FAV:
      return {
        ...state,
        flowers: state.flowers.map(flower => {
          if (flower.id === payload) {
            return { ...flower, favorite: true };
          }
          return flower;
        })
      };
    default:
      return state;
  }
};
