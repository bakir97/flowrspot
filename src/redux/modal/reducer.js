import { OPEN_MODAL, CLOSE_MODAL } from "../constants";
const initialState = {
  modal: null
};

export default (state = initialState, { type, payload }) => {
  switch (type) {
    case OPEN_MODAL:
      return { ...state, modal: payload };
    case CLOSE_MODAL:
      return { ...state, modal: payload };

    default:
      return state;
  }
};
