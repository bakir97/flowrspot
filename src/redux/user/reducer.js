import { SUCCESS_USER, LOADING_USER, ERROR_USER, SET_USER } from "../constants";
const initialState = {
  userSuccess: false,
  loading: false,
  errors: [],
  user: null
};

export default (state = initialState, { type, payload }) => {
  switch (type) {
    case SUCCESS_USER:
      return { ...state, userSuccess: payload };
    case LOADING_USER:
      return { ...state, loading: payload };
    case ERROR_USER:
      return { ...state, errors: payload };
    case SET_USER:
      return { ...state, user: payload };

    default:
      return state;
  }
};
