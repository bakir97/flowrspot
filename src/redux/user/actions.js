import axios from "axios";
import { devKeys } from "../../globals/keys";
import { SUCCESS_USER, LOADING_USER, ERROR_USER, SET_USER } from "../constants";
import { getFlowers } from "../flowers/actions";
export const registerUser = data => async dispatch => {
  try {
    dispatch(loadingUser(true));
    await axios.post(`${devKeys.api}/users/register`, data);
    dispatch(successUser(true));
    dispatch(loadingUser(false));
  } catch (error) {
    dispatch(loadingUser(false));
    dispatch(errorUser(error.response.data.error));
  }
};
export const loginUser = data => async dispatch => {
  try {
    dispatch(loadingUser(true));
    const user = await axios.post(`${devKeys.api}/users/login`, data);
    dispatch(successUser(true));

    const userData = {
      auth_token: user.data.auth_token,
      email: data.email
    };
    dispatch(login(userData));
    dispatch(getFlowers(user.data.auth_token));
    localStorage.setItem("user", JSON.stringify(userData));
    dispatch(loadingUser(false));
  } catch (error) {
    dispatch(loadingUser(false));
    dispatch(errorUser([error.response.data.error]));
  }
};
export const login = user => async dispatch => {
  try {
    const userCreds = await axios.get(`${devKeys.api}/users/me`, {
      headers: { Authorization: user.auth_token }
    });
    dispatch(setUser({ ...userCreds.data.user, email: user.email }));
  } catch (error) {
    dispatch(loadingUser(false));
    dispatch(errorUser([error.response.data.error]));
  }
};
export const logout = () => dispatch => {
  localStorage.removeItem("user");
  dispatch(getFlowers());
  dispatch(setUser(null));
};
export const successUser = payload => ({
  type: SUCCESS_USER,
  payload
});
export const setUser = payload => ({
  type: SET_USER,
  payload
});
export const loadingUser = payload => ({
  type: LOADING_USER,
  payload
});
export const errorUser = payload => ({
  type: ERROR_USER,
  payload
});
