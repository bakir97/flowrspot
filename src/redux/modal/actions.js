import { OPEN_MODAL, CLOSE_MODAL } from "../constants";
import { successUser } from "../user/actions";
export const openModal = type => dispatch => {
  dispatch(openModalAction(type));
  dispatch(successUser(false));
};
export const openModalAction = payload => ({
  type: OPEN_MODAL,
  payload
});

export const closeModal = () => async dispatch => {
  dispatch(closeModalAction());
  dispatch(successUser(false));
};
export const closeModalAction = () => ({
  type: CLOSE_MODAL,
  payload: null
});
