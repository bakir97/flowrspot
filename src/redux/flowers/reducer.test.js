import reducer from "./reducer";
import { initialState } from "./reducer";
import { successFlowers, loadingFlowers, addFavFlower } from "./actions";
describe("flowers reducer", () => {
  it("should return the initial state", () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });
  it("should handle SUCCESS_FLOWERS", () => {
    expect(
      reducer(initialState, successFlowers([{ id: 1 }]))
    ).toMatchSnapshot();
  });
  it("should handle LOADING_FLOWERS", () => {
    expect(reducer(initialState, loadingFlowers(true))).toMatchSnapshot();
  });
  it("should handle ADD_FAV", () => {
    expect(reducer(initialState, addFavFlower(1))).toMatchSnapshot();
  });
});
