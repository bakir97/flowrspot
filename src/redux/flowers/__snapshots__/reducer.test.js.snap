// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flowers reducer should handle ADD_FAV 1`] = `
Object {
  "flowers": Array [],
  "loading": false,
}
`;

exports[`flowers reducer should handle LOADING_FLOWERS 1`] = `
Object {
  "flowers": Array [],
  "loading": true,
}
`;

exports[`flowers reducer should handle SUCCESS_FLOWERS 1`] = `
Object {
  "flowers": Array [
    Object {
      "id": 1,
    },
  ],
  "loading": false,
}
`;

exports[`flowers reducer should return the initial state 1`] = `
Object {
  "flowers": Array [],
  "loading": false,
}
`;
